### ISSUE TITLE
JBR declared with `JETBRAINS` vendor in Gradle JVM Toolchain don't ship with JCEF / DCEVM (and also slight version differences)

### ISSUE THREAD

**Issue by bric3** [JetBrains/gradle-intellij-plugin#1511]
### What happened?

Declaring the jvm toolchain as

```kotlin
kotlin {
    jvmToolchain {
        languageVersion = JavaLanguageVersion.of(17)
        vendor = JvmVendorSpec.JETBRAINS
    }
}
```

and/or

```kotlin
java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(17)
        vendor = JvmVendorSpec.JETBRAINS
    }
}
```

will download a JBR without JCEF, and when running the `runIde` task and debugging, system property `java.vendor.version` reports `JBR-17.0.8.1+7-1070.2-nomod`.

However, if the vendor is changed to something else, a JBR with JCEF is being used., debugging reveal the following vendor version : JBR-17.0.7+10-829.16-jcef.

Also, I don't think the `nomod` distribution comes with DCEVM patches.

Supplementary version, why the versions differ ? I remember in the past that using slightly different minor / patch versions may reveal strange issues.

### Steps to reproduce

Configure the JVM toolchains in the relevant `build.gradle.kts`

```kotlin
kotlin {
    jvmToolchain {
        languageVersion = JavaLanguageVersion.of(17)
        vendor = JvmVendorSpec.JETBRAINS
    }
}
```

and/or

```kotlin
java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(17)
        vendor = JvmVendorSpec.JETBRAINS
    }
}
```

Set a breakpoint there : `com.intellij.ui.jcef.JBCefApp#isSupported`
Then launch the `runIde` task.

![image](https://github.com/JetBrains/gradle-intellij-plugin/assets/803621/29f32f19-ed33-4341-b66b-20c6132021f7)

Change back vendor to `JvmVendorSpec.AMAZON`, and debug, see the that JCef is supported.

![image](https://github.com/JetBrains/gradle-intellij-plugin/assets/803621/921a5fc5-1c10-4891-b9a5-1b5c174a53c3)

### Gradle IntelliJ Plugin version

1.16.1

### Gradle version

8.5

### Operating System

macOS

**Comment by bric3** [comment]
I understand that
1. we may be hitting the limit of the Gradle JVM Toolchain specification, i.e specifying other dimensions is not possible

2. this could be solved by shipping JCef or DCEVM by default via gradle JVM toolchain. But this might be debatable.

**Comment by hsz** [comment]
The issue is fixed with changes to the Disco API (see https://github.com/foojayio/discoapi/issues/93).
Gradle resolves the following JBR, which is currently the latest available: `jbrsdk_jcef-17.0.9-osx-x64-b1087.11`.
